dev/io.openliberty.jcache.internal_fat.hazelcast/fat/src/io/openliberty/jcache/internal/fat/JCacheDynamicUpdateTest.java

```diff
@@ -0,0 +1,154 @@
+     * <cacheManager
+     * uri="file:///${shared.resource.dir}/infinispan/infinispanbad.xml">
+     *
+     * <properties
+     * ... />
+     *
+     * <cachingProvider
+     * providerClass="org.infinispan.jcache.remote.JCachingProviderBad"
+     * libraryRef="InfinispanLibBad,CustomLoginLib" />
+     *
+     * </cacheManager>
+     *
+     * 1. Fix the providerClass
+     * 2. Fix the libraryRef
+     * 3. Fix the uri
+     * 4. Ensure auth cache / auth provider starts
+     *
+     * @throws Exception if the test fails for some unforeseen reason.
+     */
+    @Test
+    @CheckForLeakedPasswords(USER1_PASSWORD)
+    @AllowedFFDC(value = { "javax.cache.CacheException" })
+    public void dynamicUpdate() throws Exception {
+        /**
+         * First issue is bad cachingProvider providerClass.
+         *
+         * Look for providerClass error:
+         * CWLJC0004E: The io.openliberty.jcache.cache[AuthCache]...
+         * javax.cache.CacheException: Failed to load the CachingProvider [org.infinispan.jcache.remote.JCachingProviderBad]
+         * Caused by: java.lang.ClassNotFoundException
+         *
+         */
+        String error = "javax.cache.CacheException: Failed to load the CachingProvider \\[org.infinispan.jcache.remote.JCachingProviderBad\\]";
+        assertTrue("Should find '" + error + "' in the logs", !server1.findStringsInLogsAndTraceUsingMark(error).isEmpty());
+
+        //Update server to have correct JCachingProvider (bad libraryRef and uri)
+        setServerConfiguration(server1, "server_dynamicchange1.xml");
+
+        /**
+         * The providerClass is fixed. Look for libraryRef warning.
+         * CWWKG0033W: The value [InfinispanLibBad] specified for the reference attribute [libraryRef] was not found in the configuration.
+         */
+        error = "CWWKG0033W: The value \\[InfinispanLibBad\\] specified for the reference attribute \\[libraryRef\\] was not found in the configuration.";
+        assertTrue("Should find '" + error + "' in the logs", !server1.findStringsInLogsAndTraceUsingMark(error).isEmpty());
+
+        //Update server to have the correct library (bad uri)
+        setServerConfiguration(server1, "server_dynamicchange2.xml");
+
+        /**
+         * The providerClass and libraryRef are fixed. Look for bad uri error.
+         * CWLJC0011E: Error encountered while retrieving the AuthCache JCache: javax.cache.CacheException: Could not load configuration
+         * Caused by: java.io.FileNotFoundException: /Users/eschr/libertyGit/open-liberty/dev/build.image/wlp/usr/shared/resources/infinispan/infinispanbad.xml (No such file or
+         * directory)
+         */
+        error = "CWLJC0011E: Error encountered while retrieving the AuthCache JCache: javax.cache.CacheException: Could not load configuration";
+        assertTrue("Should find '" + error + "' in the logs", !server1.findStringsInLogsAndTraceUsingMark(error).isEmpty());
+
+        //Update to working server
```

Comment:
We don't check bad cache->cacheManagerRef and cacheManager->cachingProviderRef . I am guessing that Liberty will complain about this before it makes it to us, but we should probably check it.

I am also wondering why you didn't just use the fattest.simplicity configuration elements instead of using multiple configuration files? It might have been easier, and more clear what we were changing for each step of the test.

Reply:
I added the server config elements to fattest.simplicity and reworked the test.